spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -134,6 +138,21 @@ private RewriteDataFiles checkAndApplyFilter(
     return action;
   }
 
+  private RewriteDataFiles checkAndApplyBranch(
+      Table table, Identifier ident, RewriteDataFiles action) {
+    String branchIdent = Spark3Util.extractBranch(ident);
+    if (branchIdent != null) {
+      return action.targetBranch(branchIdent);
+    }
+    SparkWriteConf writeConf = new SparkWriteConf(spark(), table, Maps.newHashMap());
```

Comment:
Nit: can we simplify like:

```
String branchIdent = Spark3Util.extractBranch(ident);
SparkWriteConf writeConf = new SparkWriteConf(spark(), table, branch, ...);
String targetBranch = writeConf.branch();
if (targetBranch != null) {
   action.targetBranch(targetBranch);
} 
return action;
}

Reply:
Hey @szehon-ho  - im continuing the great work @zinking  did and want to finish the pr(first time contributing) ,
i changed the function  and was wondring if you think  that SparkWriteConf should receive branch as a new third argument or should i be in the  Map<String, String> writeOptions ??